Makefile

```diff
@@ -76,67 +77,80 @@ build/compiler-rt.BUILT: build/llvm.BUILT
+	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP)" \
+	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP)" \
+            -DLIBCXX_LIBDIR_SUFFIX=/wasm32-wasi \
+	    $(LLVM_PROJ_DIR)/libcxx
+	ninja $(NINJA_FLAGS) -v -C build/libcxx
+	# Do the install.
 	ninja $(NINJA_FLAGS) -v -C build/libcxx install
 	touch build/libcxx.BUILT
 
+# Flags for libcxxabi.
+LIBCXXABI_CMAKE_FLAGS = \
+    -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+    -DCMAKE_CXX_COMPILER_WORKS=ON \
+    -DCMAKE_C_COMPILER_WORKS=ON \
+    -DLIBCXXABI_ENABLE_EXCEPTIONS:BOOL=OFF \
+    -DLIBCXXABI_ENABLE_SHARED:BOOL=OFF \
+    -DLIBCXXABI_SILENT_TERMINATE:BOOL=ON \
+    -DLIBCXXABI_ENABLE_THREADS:BOOL=OFF \
+    -DLIBCXXABI_HAS_PTHREAD_API:BOOL=OFF \
+    -DLIBCXXABI_HAS_EXTERNAL_THREAD_API:BOOL=OFF \
+    -DLIBCXXABI_BUILD_EXTERNAL_THREAD_LIBRARY:BOOL=OFF \
+    -DLIBCXXABI_HAS_WIN32_THREAD_API:BOOL=OFF \
+    $(if $(patsubst 8.%,,$(CLANG_VERSION)),-DLIBCXXABI_ENABLE_PIC:BOOL=OFF,) \
+    -DCXX_SUPPORTS_CXX11=ON \
+    -DLLVM_COMPILER_CHECKED=ON \
+    -DCMAKE_BUILD_TYPE=RelWithDebugInfo \
+    -DLIBCXXABI_LIBCXX_PATH=$(LLVM_PROJ_DIR)/libcxx \
+    -DLIBCXXABI_LIBCXX_INCLUDES=$(PREFIX)/share/wasi-sysroot/include/c++/v1 \
+    -DLLVM_CONFIG_PATH=$(ROOT_DIR)/build/llvm/bin/llvm-config \
+    -DCMAKE_TOOLCHAIN_FILE=$(ROOT_DIR)/wasi-sdk.cmake \
+    -DWASI_SDK_PREFIX=$(PREFIX) \
+    -DUNIX:BOOL=ON \
+    --debug-trycompile
+
 build/libcxxabi.BUILT: build/libcxx.BUILT build/llvm.BUILT
+	# Do the build.
 	mkdir -p build/libcxxabi
-	cd build/libcxxabi; cmake -G Ninja \
-		-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
-		-DCMAKE_CXX_COMPILER_WORKS=ON \
-		-DCMAKE_C_COMPILER_WORKS=ON \
-		-DLIBCXXABI_ENABLE_EXCEPTIONS:BOOL=OFF \
-		-DLIBCXXABI_ENABLE_SHARED:BOOL=OFF \
-		-DLIBCXXABI_SILENT_TERMINATE:BOOL=ON \
-		-DLIBCXXABI_ENABLE_THREADS:BOOL=OFF \
-		-DLIBCXXABI_HAS_PTHREAD_API:BOOL=OFF \
-		-DLIBCXXABI_HAS_EXTERNAL_THREAD_API:BOOL=OFF \
-		-DLIBCXXABI_BUILD_EXTERNAL_THREAD_LIBRARY:BOOL=OFF \
-		-DLIBCXXABI_HAS_WIN32_THREAD_API:BOOL=OFF \
-		$(if $(patsubst 8.%,,$(CLANG_VERSION)),-DLIBCXXABI_ENABLE_PIC:BOOL=OFF,) \
-		-DCXX_SUPPORTS_CXX11=ON \
-		-DLLVM_COMPILER_CHECKED=ON \
-		-DCMAKE_BUILD_TYPE=RelWithDebugInfo \
-		-DLIBCXXABI_LIBCXX_PATH=$(LLVM_PROJ_DIR)/libcxx \
-		-DLIBCXXABI_LIBCXX_INCLUDES=$(PREFIX)/share/wasi-sysroot/include/c++/v1 \
-		-DLLVM_CONFIG_PATH=$(ROOT_DIR)/build/llvm/bin/llvm-config \
-		-DCMAKE_TOOLCHAIN_FILE=$(ROOT_DIR)/wasi-sdk.cmake \
-		-DLIBCXXABI_LIBDIR_SUFFIX=/wasm32-wasi \
-		-DWASI_SDK_PREFIX=$(PREFIX) \
-		-DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP) -I$(PREFIX)/share/wasi-sysroot/include" \
-		-DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP) -I$(PREFIX)/share/wasi-sysroot/include/c++/v1" \
-		-DUNIX:BOOL=ON \
-		--debug-trycompile \
-		$(LLVM_PROJ_DIR)/libcxxabi
+	cmake -B build/libcxxabi -G Ninja $(LIBCXXABI_CMAKE_FLAGS) \
+	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP)" \
+	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP)" \
+	    -DLIBCXXABI_LIBDIR_SUFFIX=/wasm32-wasi \
```

Comment:
Why are these three flags remaining inline?

Reply:
It anticipates the LTO patch, where there are different flags for LTO and non-LTO.